frontend/src/metabase/query_builder/components/expressions/ExpressionWidget.unit.spec.tsx

```diff
@@ -54,7 +56,18 @@ describe("ExpressionWidget", () => {
   });
 
   it("should trigger onChangeExpression if expression is valid", () => {
-    const { onChangeExpression } = setup();
+    const onChangeExpression = jest.fn(
+      (_name, expression, expressionClause) => {
+        expect(
+          Lib.legacyExpressionForExpressionClause(
+            createQuery(),
+            0,
+            expressionClause,
+          ),
+        ).toEqual(expression);
+      },
+    );
```

Comment:
This is really awkward but I couldn't think of a different way of testing these objects.

Maybe we should not test them?

----

I tried using `expect.objectContaining(Lib.expressionClauseForLegacyExpression(createQuery(), 0, ["+", 1, 1]))` instead of `expect.anything()` but the uuids don't match.

Extending jest with [custom matchers](https://jestjs.io/docs/expect#expectextendmatchers) is a no-go because I don't think we can easily detect that the object represents an `ExpressionClause` and say not a `OrderByClause` or anything else.

Reply:
5c75c838fab493657b6f516d0eff12c6b946d8c5